libsplinter/src/auth/oauth/mod.rs

```diff
@@ -102,11 +108,13 @@ impl OAuthClient {
     /// * `client_secret` - The GitHub OAuth client secret
     /// * `redirect_url` - The endpoint that GitHub will redirect to after it has completed
     ///   authorization
+    /// * `identity_provider` - The OAuth identity provider used to retrieve user's identity
     #[cfg(feature = "oauth-github")]
     pub fn new_github(
         client_id: String,
         client_secret: String,
         redirect_url: String,
+        identity_provider: Box<dyn IdentityProvider>,
```

Comment:
Given this is `new_github`, should this method just create the client with the github identity provider?

Reply:
Yeah, this is a good point; this method shouldn't take an identity provider, it should create the `GitHubUserIdentityProvider` internally.

Reply:
That makes sense looking at the `GitHubUserIdentityProvider`